decidim-core/spec/services/decidim/searchable_user_resource_spec.rb

```diff
@@ -43,27 +43,53 @@ module Decidim
 
     describe "Search" do
       context "when searching by User resource_type" do
-        let!(:user2) { create(:user, name: "Neil Young", organization: organization) }
-
-        it "returns User results" do
-          Decidim::Search.call("Neil", organization, resource_type: user.class.name) do
-            on(:ok) do |results_by_type|
-              results = results_by_type[user.class.name]
-              expect(results[:count]).to eq 2
-              expect(results[:results]).to match_array [user, user2]
+        let!(:user2) { create(:user, nickname: "the_loner", name: "Neil Young", organization: organization) }
+
+        context "when searching by name" do
+          it "returns User results" do
+            Decidim::Search.call("Neil", organization, resource_type: user.class.name) do
+              on(:ok) do |results_by_type|
```

Comment:
Let's DRY this up with `expect_searched_user_results("Neil", 2, [user, user2])`
(Or whatever other naming for the method that you prefer)